test/http_client_request_test.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import "dart:typed_data";
+
+import "package:http_io/http_io.dart";
+import "package:test/test.dart";
+
+Future<Null> testClientRequest(Future handler(request)) {
+  Completer<Null> completer = new Completer();
+  HttpServer.bind("127.0.0.1", 0).then((server) {
+    server.listen((request) {
+      request.drain().then((_) => request.response.close()).catchError((_) {});
+    });
+
+    var client = new HttpClient();
+    client
+        .get("127.0.0.1", server.port, "/")
+        .then((request) {
+          return handler(request);
+        })
+        .then((response) => response.drain())
+        .catchError((_) {})
+        .whenComplete(() {
+          client.close();
+          server.close();
+          completer.complete(null);
+        });
+  });
+  return completer.future;
+}
+
+Future<Null> testResponseDone() async {
+  await testClientRequest((request) {
+    request.close().then((res1) {
+      request.done.then((res2) {
+        expect(res1, equals(res2));
+      });
+    });
+    return request.done;
+  });
+}
+
+Future<Null> testBadResponseAdd() async {
+  await testClientRequest((request) {
+    request.contentLength = 0;
+    request.add([0]);
```

Comment:
FYI: these tests don't "work"

Comment out the `add` call here (which should cause the error) and the test passes.

Reply:
@zanderso – these need to be reviewed carefully –

Reply:
@kevmoo Thanks for pointing this out. It's a shame these tests weren't carefully reviewed when they were first submitted.

Reply:
agreed!